workspaces/ballerina/ballerina-core/src/rpc-types/ai-panel/index.ts

```diff
@@ -36,15 +36,19 @@ export interface AIPanelAPI {
     getFromFile: (params: GetFromFileRequest) => Promise<string>;
     getFileExists: (params: GetFromFileRequest) => Promise<boolean>;
     deleteFromProject: (params: DeleteFromProjectRequest) => void;
-    generateMappings: (params: GenerateMappingsRequest) => Promise<GenerateMappingsResponse>;
     notifyAIMappings: (params: NotifyAIMappingsRequest) => Promise<boolean>;
     stopAIMappings: () => Promise<GenerateMappingsResponse>;
     getShadowDiagnostics: (params: ProjectSource) => Promise<ProjectDiagnostics>;
     checkSyntaxError: (params: ProjectSource) => Promise<boolean>;
     clearInitialPrompt: () => void;
-    openInlineMappingChatWindow: () => void;
-    getMappingsFromModel: (params: MetadataWithAttachments) => Promise<AllDataMapperSourceRequest>;
+    openAIMappingChatWindow: (params: DataMapperModelResponse) => void;
+    generateDataMapperModel: (params: DatamapperModelContext) => Promise<DataMapperModelResponse>;
+    getTypesFromRecord: (params: GenerateTypesFromRecordRequest) => Promise<GenerateTypesFromRecordResponse>;
+    createTempFileAndGenerateMetadata: (params: CreateTempFileRequest) => Promise<ExtendedDataMapperMetadata>;
+    generateMappings: (params: MetadataWithAttachments) => Promise<AllDataMapperSourceRequest>;
+    addCodeSegmentToWorkspace: (params: CodeSegment) => void;
     addInlineCodeSegmentToWorkspace: (params: CodeSegment) => void;
+    getExpandedDMFromDMModel: (params: DMModelRequest) => Promise<ExpandedDMModelResponse>;
```

Comment:
Since this is common for both AI and rendering shall we move this to the data-mapper rpc-types?
https://github.com/VellummyilumVinoth/vscode-extension/blob/e0af6bf2d6968c7e99babbf05a8a0bec69d481d7/workspaces/ballerina/ballerina-core/src/rpc-types/data-mapper/index.ts